storage/database/shardedPersister_test.go

```diff
@@ -0,0 +1,398 @@
+		benchmarkPersister(b, keys, shardedPersisterPath, "sharded")
+	})
+}
+
+func createPersister(path string, id string) (storage.Persister, error) {
+	switch id {
+	case "simple":
+		return NewLevelDB(path, 2, 1000, 1000)
+	case "sharded":
+		shardCoordinator, _ := NewShardIDProvider(4)
+		return NewShardedPersister(path, 2, 1000, 1000, shardCoordinator)
+	default:
+		return NewLevelDB(path, 2, 1000, 10)
+	}
+}
+
+func createAndPopulatePersister(path string, entries map[string][]byte) storage.Persister {
+	db, err := createPersister(path, "simple")
+	if err != nil {
+		log.Error("createAndPopulatePersister: failed to create persister", "error", err.Error())
+		return nil
+	}
+
+	for key, val := range entries {
+		_ = db.Put([]byte(key), val)
+	}
+
+	err = db.Close()
+	if err != nil {
+		log.Error("createAndPopulatePersister: failed to close persister", "error", err.Error())
+		return nil
+	}
+
+	return db
+}
+
+func createAndPopulateShardedPersister(path string, entries map[string][]byte) storage.Persister {
+	db, err := createPersister(path, "sharded")
+	if err != nil {
+		log.Error("createAndPopulateShardedPersister: failed to create persister", "error", err.Error())
+		return nil
+	}
+
+	for key, val := range entries {
+		_ = db.Put([]byte(key), val)
+	}
+
+	err = db.Close()
+	if err != nil {
+		log.Error("createAndPopulateShardedPersister: failed to close persister", "error", err.Error())
+		return nil
+	}
+
+	return db
+}
+
+func benchmarkPersisterOneKey(b *testing.B, key []byte, path string, id string) {
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		getKey(b, key, path, id)
+	}
+}
+
+func benchmarkPersister(
+	b *testing.B,
+	keys []string,
+	path string,
+	id string,
+) {
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		getKeys(b, keys, path, id)
+	}
+}
+
+func getKeys(
+	b *testing.B,
+	keys []string,
+	path string,
+	id string,
+) {
+	b.StopTimer()
+	db, err := createPersister(path, id)
+	if err != nil {
+		log.Error("failed to create persister", "error", err.Error())
+		b.StartTimer()
+		return
+	}
+	b.StartTimer()
+
+	defer func() {
+		b.StopTimer()
+		err := db.Close()
+		if err != nil {
+			log.Error("failed to close persister", "error", err.Error())
+		}
+		b.StartTimer()
+	}()
+
+	maxRoutines := make(chan struct{}, 400)
+	wg := sync.WaitGroup{}
+	wg.Add(len(keys))
+
+	for _, key := range keys {
+		maxRoutines <- struct{}{}
+		go func(key string) {
+			defer func() {
+				<-maxRoutines
+				wg.Done()
+			}()
+
+			_, err = db.Get([]byte(key))
+			if err != nil {
+				log.Info("key NOT found", "key", key)
+			}
+			require.Nil(b, err)
+		}(key)
+	}
+
+	wg.Wait()
+}
+
+func getKey(
+	b *testing.B,
+	key []byte,
+	path string,
+	id string,
+) {
+	b.StopTimer()
+	db, err := createPersister(path, id)
+	if err != nil {
+		log.Error("failed to create persister", "error", err.Error())
+		b.StartTimer()
+		return
+	}
+	b.StartTimer()
+
+	go func(key []byte) {
+		defer func() {
+			b.StopTimer()
```

Comment:
Extract L358-L363 in a new func and use it in both `getKey` and `getKeys`

Reply:
done